src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CaseRepository.java

```diff
@@ -24,6 +25,8 @@
 
     void saveDefendantResponse(Claim claim, String defendantEmail, Response response, String authorization);
 
+    void saveClaimantResponse(long claimId, ClaimantResponse response, String authorization);
```

Comment:
same here, prefer not to use DB generated id in contract here as won't work for CCD. May be externalId (UUID) is alternative

Reply:
why wouldn't it work on ccd?

Reply:
I've checked, and id is correct here.
On an update to CCD we need the ID so that we don't make any additional calls to retrieve it,
the settlement code does the same as this although it passed the whole claim then gets the ID straight out and passes it along to the service.

The whole object is not needed here

Reply:
We don't need to pass whole object, just replace DB generated id with externalId, would be like:
```
 void saveClaimantResponse(String externalId, ClaimantResponse response, String authorization);
```

Reply:
no no no, then you have to call CCD again to retrieve the claim object you had just before.
We need the ID for the next ccd call. I've passed the required information

Reply:
There will be no additional call to CCD or any repository. We are also passing `externalId` to the API endpoint. 
```
@Transactional(transactionManager = "transactionManager")
    public void save(
        String externalId,
        String claimantId,
        ClaimantResponse response,
        String authorization
    ) {
        Claim claim = claimService.getClaimByExternalId(externalId, authorization);
        claimantResponseRule.assertCanBeRequested(claim, claimantId);

        caseRepository.saveClaimantResponse(claim.getId(), response, authorization); 
        // The above will change to as below, no additional call to retrieve claim
       caseRepository.saveClaimantResponse(claim.getExternalId(), response, authorization); 

        appInsights.trackEvent(getAppInsightsEvent(response), claim.getReferenceNumber());
    }
```

Reply:
discussed on slack, TLDR id is needed for calling to CCD. The contract here is correct